### PR TITLE
Make sharpness output channel nums as input

### PR DESCRIPTION
**Pull request by shijianjian**
### Description
#781 

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before ?
- [ ] **Pass all tests**: did you test in local ? `make test`
- [ ] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [ ] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

File: kornia/enhance/adjust.py [hunk 1]
```diff
@@ -406,20 +406,30 @@ def sharpness(input: torch.Tensor, factor: Union[float, torch.Tensor]) -> torch.
...
+    if len(factor.size()) != 0:
+        assert input.size(0) == factor.size(0), \
+            f"Input batch size shall match with factor size if 1d array. Got {input.size(0)} and {factor.size(0)}"
+
     kernel = torch.tensor([
         [1, 1, 1],
         [1, 5, 1],
         [1, 1, 1]
-    ], dtype=input.dtype, device=input.device).view(1, 1, 3, 3).repeat(3, 1, 1, 1)
+    ], dtype=input.dtype, device=input.device).view(1, 1, 3, 3).repeat(input.size(1), 1, 1, 1)
```
Comment (edgarriba) [inline kornia/enhance/adjust.py:432]: Add proper testing

File: test/enhance/test_adjust.py [hunk 2]
```diff
@@ -742,3 +742,57 @@ def build_input(channels, depth, height, width, bs=1, row=None):
...
+             [0.0223, 0.8338, 0.2939],
+             [0.5185, 0.6977, 0.8000]]]], device=device, dtype=dtype)
+
+        assert_allclose(TestSharpness.f(inputs, 0.8), expected, rtol=1e-4, atol=1e-4)
+
+    def test_gradcheck(self, device, dtype):
+        bs, channels, height, width = 2, 3, 4, 5
+        inputs = torch.ones(bs, channels, height, width, device=device, dtype=dtype)
+        inputs = utils.tensor_to_gradcheck_var(inputs)
+        assert gradcheck(TestSharpness.f, (inputs, 0.8), raise_exception=True)
```
Comment (edgarriba) [inline test/enhance/test_adjust.py:798]: @shijianjian Could you try with torchscript too?

File: test/enhance/test_adjust.py [hunk 3]
```diff
@@ -742,3 +742,57 @@ def build_input(channels, depth, height, width, bs=1, row=None):
...
+        inputs = torch.ones(bs, channels, height, width, device=device, dtype=dtype)
+
+        assert TestSharpness.f(inputs, 0.8).shape == torch.Size([bs, channels, height, width])
+
+    def test_equalize(self, device, dtype):
+        torch.manual_seed(0)
+
+        inputs = torch.rand(1, 3, 3).to(device=device, dtype=dtype)
+
+        expected = torch.tensor([
```
Comment (edgarriba) [inline test/enhance/test_adjust.py:772]: Could you add a comment about how did you get those numbers. Are they from opencv?

File: kornia/enhance/adjust.py [hunk 4]
```diff
@@ -406,47 +409,72 @@ def sharpness(input: torch.Tensor, factor: Union[float, torch.Tensor]) -> torch.
...
+    Example:
+        >>> _ = torch.manual_seed(0)
+        >>> sharpness(torch.randn(1, 1, 5, 5), 0.5)
+        tensor([[[[-1.1258, -1.1524, -0.2506, -0.4339,  0.8487],
+                  [ 0.6920, -0.1580, -1.0576,  0.1765, -0.1577],
+                  [ 1.4437,  0.1998,  0.1799,  0.6588, -0.1435],
+                  [-0.1116, -0.3068,  0.8381,  1.3477,  0.0537],
+                  [ 0.6181, -0.4128, -0.8411, -2.3160, -0.1023]]]])
     """
     input = _to_bchw(input)
```
Comment (edgarriba) [inline kornia/enhance/adjust.py:422]: this is not related to this PR but shouldn't we accept only bchw to make it seamless to the rest of the library?

File: test/enhance/test_adjust.py [hunk 5]
```diff
@@ -742,3 +742,86 @@ def build_input(channels, depth, height, width, bs=1, row=None):
...
 
         return batch
+
+
+class TestSharpness(object):
+
+    f = kornia.enhance.sharpness
+
+    @pytest.mark.parametrize("channels", [1, 3, 5])
+    def test_shape(self, channels, device, dtype):
```
Comment (edgarriba) [inline test/enhance/test_adjust.py:752]: I guess you can merge this two tests into one. Would be great to inherit from BaseTester and have same test names. Recently learnt that we can filter by test names which will make easier later for the ci. See: https://github.com/kornia/kornia/pull/787/files#diff-fd3c5292dc69298c69ca985ded96d320072ac2625604af20c5bbbef1cdaff562R26

**Review by edgarriba — CHANGES_REQUESTED** [review]

File: kornia/enhance/adjust.py (hunk 4 above)
Comment (shijianjian) [reply kornia/enhance/adjust.py:422]: Yes. It makes sense. I will send another PR for this.

**Review by edgarriba — APPROVED** [review]
LGTM
